### ISSUE TITLE
Batch insert generated keys

### ISSUE THREAD

**Issue by bardam** [scalikejdbc/scalikejdbc#487]
Is there a way to get all the generated keys of a batch insert?

**Comment by seratch** [comment]
It's not possible for now because JDBC Batch API returns only an array of update count. https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#executeBatch()

**Comment by bardam** [comment]
take a look here:
[https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#getGeneratedKeys()](https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#getGeneratedKeys())

I managed to do it with the following code:
```scala
// val session: DBSession = ...
val st = session.connection.createStatement()
st.addBatch(s"insert into person (first_name, last_name) values ('fn1', 'ln1')")
st.addBatch(s"insert into person (first_name, last_name) values ('fn2', 'ln2')")
st.executeBatch()
val rs = st.getGeneratedKeys
while (rs.next()) {
  println(rs.getInt(1))
}

the table I'm inserting into has auto generated key
```

**Comment by seratch** [comment]
I see, it's my misunderstanding. Thank you for your pointing it out. Adding new API which returns generated keys to batch operation may be worth considering.

**Comment by seratch** [comment]
@bardam I've implemented this feature at #492. If you have some comments, write them there.

**Comment by cppexpert** [comment]
@seratch Could you please demonstrate how to get generated keys when using `batch` insert in QueryDSL ?

Consider following example
```scala
val names = articleColumns.columns.map(_ -> sqls.?)
val values = items.map(Generic[Article].to(_).values.map(toSqlTypedHlist).toList)
withSQL {
  insert.into(Article).namedValues(names: _*)
}.batch(values: _*)().sum
```

**Comment by cppexpert** [comment]
Ah... It wasn't merged into `master` yet, isn't it ? When are you planning on merging it ?

**Comment by seratch** [comment]
@cppexpert My PR #492 to fix this issue will be merged soon and upcoming 2.4 which includes the feature release will come at least in this month.
